hs_core/models.py

```diff
@@ -86,7 +87,11 @@ def can_add(self, request):
     def can_delete(self, request):
```

Comment:
@selimnairb @pkdash Are there test cases written already to test authorize() function? I cannot find any test cases for authorize() function in our current code base. Since authorize() function uses hs_access_control functions directly to check for authorization and hs_access_control has already been well-tested, I am thinking we may not need to test this function although I agree it'd be nice to test all the combinations of different input parameters to this function. Since the new behavior for published and immutable flags are covered in hs_access_control functions, I don't need to change this authorize() much except when full is true, we should call uaccess.can_delete_resource() rather than owns_resource() to account for published and immutable flags. @selimnairb If we don't have test cases for authorize() already and we want to add them, could you create an issue for it so we can tackle it separately from this PR? Otherwise, if we already have test cases, let me know where they are and I can augment test cases as needed. Let me know how you want to proceed with this.